### PR TITLE
List Emercoin (EMC)

### PR DESCRIPTION
**Pull request by maxihatop**
- Official project URL: https://emercoin.com/en
- Official block explorer URL: https://explorer.emercoin.com/

### CODE REVIEW

**Comment by p3yot3** [comment]
Listing coins that have multiple unanswered scam accusations against the devs on BCT is not desirable.

**Comment by maxihatop** [comment]
Can you specify any really proven fact, when Emercoin cheated someone, people or organization?
Emercoin exists since 2013, and during lifetime of our projects many other coins appeared and vanished. We developed unique blockchain-based technologies, you can see web-site for details. Really, I personally do not answering to internet-throlls, because of those accusations just defamation. Read those accusations more detailed, and you will see, there is "a lie over a lie".
Please, disclose any fact, when I am, or anyone from Emercoin did something unethical or criminal.

**Comment by ghost** [comment]
@p3yot3 
Writing what you write without even providing a single link supporting it is not very serious (and is borderline behaviour imo).

**Comment by p3yot3** [comment]
> 
> 
> @p3yot3
> Writing what you write without even providing a single link supporting it is not very serious (and is borderline behaviour imo).

As this is not a forum I didn't want to clutter this excellent projects repo, especially as all links to said scam accusations with proof are available on the Emercoin BCT thread for all to see & have been for some time, but simply searching for Emercoin in the scam thread (which is what I did) will reveal all. The Emercoin "team" (Oleg Khovayko (maxihatop), Jevgeni Shumilov, Stan Polozov etc) are fully aware of this & the scam accusations but have chosen to ignore them & any questions from forum members regarding their various scams (Hashcoins, Hashflare, Polybius, Chronobank, Hamster to name a few I found) - which can only be regarded as an admission of guilt. It is difficult to understand how they can be described as "a lie over a lie" when they have their name & photos on the scam projects ANN threads as part of the team responsible for these various proven scams. Simply calling the people who uncovered their scams trolls doesn't wash. Anyway, here are a few links in case you missed them, there are many, many more:

https://bitcointalk.org/index.php?topic=362513.msg51710874#msg51710874
https://bitcointalk.org/index.php?topic=1970796.0
https://bitcointalk.org/index.php?topic=417623.msg19333790#msg19333790

Bisq listing coins from such prolific scammers will not only damage it's excellent reputation but will attract the wrong kind of attention. The anonymous nature of Bisq appeals to scammers, so diligence & research should be done on the people behind such coins as Emercoin, especially as it has been refused listing on many legit exchanges (Poloniex for one) & recently delisted from others (Bittrex being the latest) - there is a reason for this & it's not because they are legit.

I will not be posting about this again or replying questions, these are my observations based on facts that are available to everyone, should they choose to look for them.

**Comment by sysmanalex** [comment]
Dear @p3yot3 ! 
Emercoin is platform same as ETH, it offering services ssh/ssl/pki/dns... 
all these hashcoins/hashflare/polybius/chronobank - is separate project & has separate teams members, they was just planing to use services ssl/pki -  there is here relation ? or real facts about emercoin ?

**Comment by p3yot3** [comment]
> 
> 
> Dear @p3yot3 !
> Emercoin is platform same as ETH, it offering services ssh/ssl/pki/dns...
> all these hashcoins/hashflare/polybius/chronobank - is separate project & has separate teams members, they was just planing to use services ssl/pki - there is here relation ? or real facts about emercoin ?

Alex Petrov, I read about you here:

https://bitcointalk.org/index.php?topic=1942398.msg27930110#msg27930110

Will the rest of the Emercoin scam team be posting here also?

> Emercoin making a scammy come back now that Bitfury and George Kikvadze & Alex Petrov are behind it.  Hopefully it can be crushed.

**Comment by sysmanalex** [comment]
@p3yot3 here is a proof - its totally different teams & projects - no overlapping at all.  
https://polybius.io/#team
https://www.hashcoins.com/management
https://hashflare.io/team/
https://polybius.io/#team
and  https://emercoin.com/team - different ppl, different project - try to realize emercoin is platform, there are no pre-mine, no tokens, its not ICO. its open source project for services.

**Comment by sysmanalex** [comment]
@p3yot3 
Please study the facts... - emercoin is platform & has different team, all these listed projects ...  who was planing to use emercoins - as bad consequence for emercoin image - someone mix different projects with platform all together. 
- emercoin as a lot of other good cases. please study to get better understanding of background http://emercoin.com/news 
there nothing to answer for team, then they asked about some external projects there they are not involved, this FUD just spreading because someone just mix all together and its sad.

**Comment by maxihatop** [comment]
Posts from @p3yot3 is a great example of mixing things together. All those 3 posts do not shows any facts, what exactly - Emercoin as blockchain platform did wrong!

All 3 demonstrated just a simple foundless accusation:
Hashcoin/Hashflare/Polybius/Chronobank - was using Emercoin blockchain, or just promising to use some of Emercoin blockchain functions for their projects, and because of that - the Emercoin is scam?!
There is a different projects, different teams… Please do not mix all together!

- Emercoin is open source platform, providing a lot of useful services like ssl/ssh/dns/pki/timestamping - all upper listed projects are just using some of services. 
 - Emercoin team members in not involved in these projects, at all - same was as ETH is not responsible for any ICO fails, or no-one blaming TCP/IP protocol for been hacked.   

Yes, we shared our technologies to many projects, including projects above. But, we do not responsible for all activities, where is using our technologies. As same as Github is not responsible for @p3yot3's posts.

Again: please, demonstrate facts - what Emercoin did wrong, and what team did wrong. Otherwise, without a real facts. your accusations just unsubstantiated.

**Comment by p3yot3** [comment]
Thanks to the internet we can see that Oleg Khovayko (maxihatop) was a member of Polybius - it was removed when the connections were made by the crypto community:

https://web.archive.org/web/20170420094524/http://www.polybius.io/

If you guys had put half as much effort into your numerous scam accusations as you have here you might have convinced people, but thanks to internet archive, you are only confirming your own lies.

**Comment by maxihatop** [comment]
@p3yot3,
You false claimed in the your post "Oleg Khovayko (maxihatop) was a member of Polybius".
I was not a **team member**, I was an **technical consultant**. Please, read carefully link, provided by you above. And who is lies?

**Comment by p3yot3** [comment]
Thank you for confirming that you "advise" scams Oleg, not that it needed any confirming because here's another one:

https://bitcointalk.org/index.php?topic=3098967.msg32143800#msg32143800

...looks like you & Stan Polosov (Emercoin CMO) forgot to get your photos removed from that one though.

**Comment by maxihatop** [comment]
Once again - I never was a team member of their project, I was an technical consultant, technical advisor for ssl/auth implementation, they asking me technical questions about Emercoin - I was answering... They took some emer logo, they used my picture - but I'm not a part of their project. not it sad story.

**Comment by p3yot3** [comment]
Thank you for confirming that you advised on another scam project Oleg. You have quite the reputation for being a scam advisor.

**Comment by sysmanalex** [comment]
> Thanks to the internet we can see that Oleg Khovayko (maxihatop) was a member of Polybius - it was removed when the connections were made by the crypto community:
> https://web.archive.org/web/20170420094524/http://www.polybius.io/
> 
> If you guys had put half as much effort into your numerous scam accusations as you have here you might have convinced people, but thanks to internet archive, you are only confirming your own lies.

1. Again you misrepresenting facts, saying "member" then Oleg was **external consultant** - can you fill the difference? ...he doesn't know about nature of project, just help by answering on questions like anyone else. - he doesn't has time to read all bitcointalk threads, not related to emercoin - all this fud was in different threads. no direct questions or messages.

2. I see massive black PR & hate - not facts, or inverting facts, or ignoring answers. just mention that they planing to use emercoin ssl/auth as good solution, doesn't make emercoin bad.
you reading from walls & repeating fud posted by someone as their version, without analyzing.

3. Emercoin is not ICO, doesn't has any pre-mine or pre-generated tokens.
  emercoin don't sell you anything,  emercoin doesn't ask anything from you... emer its open source project, offer services, you may use then or you may ignore it...very simple.

please pay time, study project, study history - emercoin did long time ago SSH/SSL/PKI on blockchain - even before all listed projects, DNS, ENUM E.164, TTS(timestamp), NVS, notary, diplomas, vpn... emc offering service platform - check that was done...  study subject, before spreading FUD and inverting facts on each step, its your imho only here.

emercoin developing product & offer services, we doesn't sell you anything, no tokens, no ico's.

**Comment by p3yot3** [comment]
Please address your scam accusations on BCT - not here.

File: assets/src/main/java/bisq/asset/coins/Emercoin.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+package bisq.asset.coins;
+
+import bisq.asset.Coin;
+import bisq.asset.DefaultAddressValidator;
+
+public class Emercoin extends Coin {
+
+    public Emercoin() {
+        super("Emercoin", "EMC", new DefaultAddressValidator());
```
Comment (blabno) [inline assets/src/main/java/bisq/asset/coins/Emercoin.java]: Do not use deprecated validator.

File: assets/src/test/java/bisq/asset/coins/EmercoinTest.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ * along with Bisq. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package bisq.asset.coins;
+
+import bisq.asset.AbstractAssetWithDefaultValidatorTest;
+
+public class EmercoinTest extends AbstractAssetWithDefaultValidatorTest {
+
+    public EmercoinTest() {
```
Comment (blabno) [inline assets/src/test/java/bisq/asset/coins/EmercoinTest.java:26]: Test class is missing test methods.

File: core/src/main/java/bisq/core/locale/CurrencyUtil.java [hunk 3]
```diff
@@ -131,6 +131,7 @@ public static void setBaseCurrencyCode(String baseCurrencyCode) {
         result.add(new CryptoCurrency("BEAM", "Beam"));
         result.add(new CryptoCurrency("DASH", "Dash"));
         result.add(new CryptoCurrency("DCR", "Decred"));
+        result.add(new CryptoCurrency("EMC", "Emercoin"));
```
Comment (blabno) [inline core/src/main/java/bisq/core/locale/CurrencyUtil.java]: Do not touch this file.

**Review by blabno — CHANGES_REQUESTED** [review]
NACK per comments

**Comment by maxihatop** [comment]
Hi blabno,
I need your suggestion.
I see, BISQ requires to submit "Add currency" with a single pull request. I do not know, how to correctly apply your changes, what must I to do? I see following options:
1. Cancel this pull request, and create another one, using your recommendation?
2. Edit/commit files within my repository? If so, how to transfer them into this pull request?

I would like to apply your changes, but please, provide me a guide, ho to correctly accept them, and comply BISQ requirements.

In addition: Can you provide me good example of another bitcoin-based crypto in the BISQ repositary, which I can use as example/template to correctly update request? I used Namecoin as an example, but you write, it is a wrong "deprecated validator".

**Comment by blabno** [comment]
Squash/amend commits and push the same branch with force.

**Comment by maxihatop** [comment]
Blanbo,
Per your suggestion, corrected files and pushed with 'git push -f'
Hope, everything is correct right now.

**Review by blabno — APPROVED** [review]

**Review by blabno — CHANGES_REQUESTED** [review]
Please squash into a single commit.

**Comment by blabno** [comment]
@p3yot3 I don't want to make political decisions here. The PR conforms to our guideline so I will acknowledge it. 
Please create proposal on de-listing this asset using DAO if you have strong feelings against it and the community will decide.

**Comment by p3yot3** [comment]
@blabno Agreed. Will do.

**Comment by maxihatop** [comment]
@blabno, squash done.

**Review by blabno — APPROVED** [review]
ACK

**Pull request merged**